### PR TITLE
Feat uupdate my profile page

### PR DESCRIPTION
**Pull request by DiLiUA**

### CODE REVIEW

File: src/my-profile/my-profile.component.ts [hunk 1]
```diff
@@ -59,11 +96,15 @@ export class MyProfileComponent implements OnDestroy {
 
     const updateUserSubscription = this.userService.updateUser(rqstObj)
       .subscribe(res => {
+        this.birthdayConfig.isChanged = false;
```
Comment (Ugmaxie) [inline src/my-profile/my-profile.component.ts]: dateOfBirth

File: src/my-profile/my-profile.component.ts [hunk 2]
```diff
@@ -59,11 +96,15 @@ export class MyProfileComponent implements OnDestroy {
...
         this.currentUser.username = this.changableData.username;
         this.currentUser.email = this.changableData.email;
         this.currentUser.biography = this.changableData.biography;
+        this.currentUser.birthday = this.changableData.birthday;
```
Comment (Ugmaxie) [inline src/my-profile/my-profile.component.ts]: dateOfBirth

File: src/my-profile/my-profile.component.ts [hunk 3]
```diff
@@ -105,14 +146,20 @@ export class MyProfileComponent implements OnDestroy {
   }
 
   setChangableData(): void {
+    this.birthdayConfig.currentValue = this.currentUser.birthday;
```
Comment (Ugmaxie) [inline src/my-profile/my-profile.component.ts]: dateOfBirth

File: src/my-profile/my-profile.component.ts [hunk 4]
```diff
@@ -105,14 +146,20 @@ export class MyProfileComponent implements OnDestroy {
...
+
     const changableData: ChangableData = {
       username: this.currentUser.username,
+      firstName: this.currentUser.firstName,
+      lastName: this.currentUser.lastName,
+      birthday: this.currentUser.birthday,
```
Comment (Ugmaxie) [inline src/my-profile/my-profile.component.ts]: dateOfBirth

File: src/my-profile/my-profile.component.ts [hunk 5]
```diff
@@ -141,7 +188,23 @@ export class MyProfileComponent implements OnDestroy {
     this.subscriptionManager.add(uploadAvatarSubscription);
   }
 
+  changeBirthday(date: Date): void {
```
Comment (Ugmaxie) [inline src/my-profile/my-profile.component.ts]: changeDateOfBirth

File: src/user-service/user.model.ts [hunk 6]
```diff
@@ -1,3 +1,7 @@
+import { environment } from '../environments/environment';
+import { RouterLinks } from '../enums';
+import { City, Country } from '../shared/models';
+
 interface UserInterface {
   _id?: string;
```
Comment (Ugmaxie) [inline src/user-service/user.model.ts:6]: is it really "?"

**Review by Ugmaxie — CHANGES_REQUESTED** [review]

File: src/user-service/user.model.ts (hunk 6 above)
Comment (DiLiUA) [reply src/user-service/user.model.ts:6]: yes it is

**Review by Ugmaxie — APPROVED** [review]

**Pull request merged**